### PR TITLE
[AV-325] Implement RASAero Import class internals

### PR DESCRIPTION
**Pull request by AyberkY**
This PR implements the internals of the `RASAeroImport` class. Namely the parsing of the RASAero data into a lookup table, extracting metadata about the aerodynamic data we have, and the lookup function itself.

The PR _does not_ integrate the data into the physics yet. Only implements the lookup functionality of the class.

The data achieved from the class will be integrated in the PhysicsEngine in [[AV-293]](https://trello.com/c/TYB336US/293-av-293-implement-aero-force-and-moment-calculations-using-data-obtained-from-rasaero)

### CODE REVIEW

**Comment by AyberkY** [comment]
[[AV-325] Write C++ interface to ingest aggregated RASAero data](https://trello.com/c/VnH64TrX/325-av-325-write-c-interface-to-ingest-aggregated-rasaero-data)

File: src/RASAeroImport.cpp [hunk 1]
```diff
@@ -13,8 +15,250 @@
 
 #include <rapidcsv.h>
 
+#include <cmath>
+#include <iostream>
+#include <set>
```
Comment (NPIPHI) [inline src/RASAeroImport.cpp:20]: you don't use set anywhere

**Review by NPIPHI — APPROVED** [review]
Interpolating code looks good, a few style things but looks good to merge

File: src/RASAeroImport.cpp [hunk 2]
```diff
@@ -13,8 +15,250 @@
...
+    std::cout << "### [RASAeroImport ctor parsing metadata]:" << std::endl;
+    std::cout << "mach_instances = " << mach_number_instances_ << std::endl;
+    std::cout << "mach_fidelity = " << mach_number_fidelity_ << std::endl;
+    std::cout << "alpha_instances = " << alpha_instances_ << std::endl;
+    std::cout << "alpha_fidelity = " << alpha_fidelity_ << std::endl;
+    std::cout << "protuberance_instances = " << protuberance_instances_
+              << std::endl;
+    std::cout << "protuberance_fidelity = " << protuberance_fidelity_
+              << std::endl;
+#endif
```
[the remark this answers is not in the document]
Comment (AyberkY) [reply src/RASAeroImport.cpp]: fixed

File: src/RASAeroImport.cpp [hunk 3]
```diff
@@ -13,8 +15,250 @@
...
+ * - mach_number_instances_: The number of unique mach number values that exist
+ *       in the lookup table
+ * - mach_number_fidelity_: The distance between two mach number instances
+ *
+ */
+void RASAeroImport::set_mach_number_params() {
+    auto column = aero_table_.col(0);
+    auto vec = std::vector<double>(column.begin(), column.end());
+    sort(vec.begin(), vec.end());
+    vec.erase(unique(vec.begin(), vec.end()), vec.end());
```
[the remark this answers is not in the document]
Comment (AyberkY) [reply src/RASAeroImport.cpp]: fixed

File: src/RASAeroImport.cpp [hunk 4]
```diff
@@ -13,8 +15,250 @@
...
+    mach_number_instances_ = vec.size();
+    mach_number_fidelity_ = fabs(vec[0] - vec[1]);
```
[the remark this answers is not in the document]
Comment (AyberkY) [reply src/RASAeroImport.cpp]: fixed

File: src/RASAeroImport.cpp [hunk 5]
```diff
@@ -13,8 +15,250 @@
...
+ * - alpha_fidelity_: The distance between two alpha instances
+ *
+ */
+void RASAeroImport::set_alpha_params() {
+    auto column = aero_table_.col(1);
+    auto vec = std::vector<double>(column.begin(), column.end());
+    sort(vec.begin(), vec.end());
+    vec.erase(unique(vec.begin(), vec.end()), vec.end());
+    alpha_instances_ = vec.size();
+    alpha_fidelity_ = fabs(vec[0] - vec[1]);
```
[the remark this answers is not in the document]
Comment (AyberkY) [reply src/RASAeroImport.cpp]: fixed

File: src/RASAeroImport.cpp [hunk 6]
```diff
@@ -13,8 +15,250 @@
...
+ * - protuberance_fidelity_: The distance between two protuberance instances
+ *
+ */
+void RASAeroImport::set_protuberance_params() {
+    auto column = aero_table_.col(2);
+    auto vec = std::vector<double>(column.begin(), column.end());
+    sort(vec.begin(), vec.end());
+    vec.erase(unique(vec.begin(), vec.end()), vec.end());
+    protuberance_instances_ = vec.size();
+    protuberance_fidelity_ = fabs(vec[0] - vec[1]);
```
[the remark this answers is not in the document]
Comment (AyberkY) [reply src/RASAeroImport.cpp]: fixed

File: src/RASAeroImport.cpp [hunk 7]
```diff
@@ -13,8 +15,250 @@
 
 #include <rapidcsv.h>
 
+#include <cmath>
+#include <iostream>
+#include <set>
 #include <string>
 
+#include "Eigen/src/Core/Matrix.h"
```
[the remark this answers is not in the document]
Comment (AyberkY) [reply src/RASAeroImport.cpp]: fixed

File: src/RASAeroImport.cpp (hunk 1 above)
Comment (AyberkY) [reply src/RASAeroImport.cpp]: removed

**Pull request merged**